packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -137,24 +146,44 @@ class RequestManager {
     required UnwrapCallback<T, R> unwrap,
     required SerializeCallback<R> serialize,
     required DeserializeCallback<R> deserialize,
+    GetCacheParameters? getCacheParameters,
     bool disableTimeout = false,
     Duration timeLimit = kDefaultTimeout,
     int retries = 0,
   }) async {
     final key = '${account.id}-$cacheKey';
 
-    Future<void>? cacheFuture;
     if (retries == 0) {
       // emit loading state with the current value if present
       final value = subject.valueOrNull?.asLoading() ?? Result.loading();
       subject.add(value);
 
-      cacheFuture = _emitCached(
+      final cachedValue = await _emitCached(
```

Comment:
Retrieving from cache can be expensive. Especially when it is large binary data (retrieving from sqlite and passing into the dart code via ffi).

I suggest a separate method to retrieve the cache parameters and only query the data if still valid.

Reply:
Good point, I didn't think about that!
I guess then the problem with the slow cache is no more?

Reply:
I have not done a performance analysis but I think that we where really favoring the network request.
But even if the entire time will be a bit slower on low end device we must also consider the saved request on the server.

Let's at least try this and if needed I can do a performance check later on